cloudsql/sqlserver/database-sql/cloudsql.go

```diff
@@ -0,0 +1,302 @@
+	}
+
+	var voteDiffStr string = voteDiff(int(math.Abs(float64(tabVotes) - float64(spaceVotes)))).String()
+
+	latestVotesCast, err := recentVotes()
+	if err != nil {
+		return templateData{}, fmt.Errorf("recentVotes: %v", err)
+	}
+	return templateData{tabVotes, spaceVotes, voteDiffStr, latestVotesCast}, nil
+
+}
+
+// showTotals renders an HTML template showing the current vote totals.
+func showTotals(w http.ResponseWriter, r *http.Request) error {
+
+	totals, err := currentTotals()
+	if err != nil {
+		return fmt.Errorf("currentTotals: %v", err)
+	}
+	err = parsedTemplate.Execute(w, totals)
+	if err != nil {
+		return fmt.Errorf("Template.Execute: %v", err)
+	}
+	return nil
+}
+
+// saveVote saves a vote passed as http.Request form data.
+func saveVote(w http.ResponseWriter, r *http.Request) error {
+	if err := r.ParseForm(); err != nil {
+		return fmt.Errorf("Request.ParseForm: %v", err)
+	}
+
+	var team string
+	if teamprop, ok := r.Form["team"]; ok {
+		team = teamprop[0]
+	} else {
+		return fmt.Errorf("team property missing from form submission")
+	}
+
+	// [START cloud_sql_sqlserver_databasesql_connection]
+	sqlInsert := "INSERT INTO votes (candidate, time_cast) VALUES (?, GETDATE())"
+	if team == "TABS" || team == "SPACES" {
+		if _, err := db.Exec(sqlInsert, team); err != nil {
+			fmt.Fprintf(w, "unable to save vote: %s", err)
+			return fmt.Errorf("DB.Exec: %v", err)
+		} else {
+			fmt.Fprintf(w, "Vote successfully cast for %s!\n", team)
+		}
+	}
+	return nil
+	// [END cloud_sql_sqlserver_databasesql_connection]
+}
+
+// mustGetEnv is a helper function for getting environment variables.
+// Displays a warning if the environment variable is not set.
+func mustGetenv(k string) string {
+	v := os.Getenv(k)
+	if v == "" {
+		log.Printf("Warning: %s environment variable not set.\n", k)
+	}
+	return v
+}
+
+// initSocketConnectionPool initializes a Unix socket connection pool for
+// a Cloud SQL instance of SQL Server.
+func initSocketConnectionPool() (*sql.DB, error) {
+	// [START cloud_sql_sqlserver_databasesql_create_socket]
+	var (
+		dbUser                 = mustGetenv("DB_USER")
+		dbPwd                  = mustGetenv("DB_PASS")
+		instanceConnectionName = mustGetenv("INSTANCE_CONNECTION_NAME")
+		dbName                 = mustGetenv("DB_NAME")
+	)
+
+	var dbURI string
+	dbURI = fmt.Sprintf("%s:%s@unix(/cloudsql/%s)/%s", dbUser, dbPwd, instanceConnectionName, dbName)
+
+	// dbPool is the pool of database connections.
+	dbPool, err := sql.Open("mssql", dbURI)
+	if err != nil {
+		return nil, fmt.Errorf("sql.Open: %v", err)
+	}
+
+	// [START_EXCLUDE]
+	configureConnectionPool(dbPool)
+	// [END_EXCLUDE]
+
+	return dbPool, nil
+	// [END cloud_sql_sqlserver_databasesql_create_socket]
+}
+
+// initTcpConnectionPool initializes a TCP connection pool for a Cloud SQL
+// instance of SQL Server.
+func initTcpConnectionPool() (*sql.DB, error) {
+	// [START cloud_sql_sqlserver_databasesql_create_tcp]
+	var (
+		dbUser    = mustGetenv("DB_USER")
```

Comment:
Seems like it should be an error if one of these isn't set? Or is logging enough?

Reply:
+1 - `log.Fatalf` is probably preferred.